src/support_code_library_builder/define_helpers.js

```diff
@@ -105,14 +106,28 @@ function getDefinitionLineAndUri(cwd) {
 }
 
 export function defineParameterType(builder) {
-  return ({ regexp, transformer, typeName }) => {
+  return ({
+    name,
+    typeName,
+    regexp,
+    transformer,
+    useForSnippets,
+    preferForRegexpMatch
+  }) => {
+    const getTypeName = deprecate(
+      () => typeName,
+      'Use name instead of typeName'
```

Comment:
Does this deprecation notice give enough information to know the context if a user sees it? I think it might be worth prefixing it with: defineParameterType